### PR TITLE
LibGL: Restructure code layout

### PR DESCRIPTION
**Pull request by GMTA**
A big QoL restructure of LibGL.

* Collect all of the OpenGL API functions in `GLAPI.cpp`;
* Split up `GLContext.cpp` into multiple code units, allowing for faster recompilation when an implementation is changed.

No functional changes.

### CODE REVIEW

File: Userland/Libraries/LibGL/Lighting.cpp [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+ * Copyright (c) 2021, Stephan Unverwerth <s.unverwerth@serenityos.org>
+ * Copyright (c) 2022, Jelle Raaijmakers <jelle@gmta.nl>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Debug.h>
+#include <LibGL/GLContext.h>
+
+extern GL::GLContext* g_gl_context;
```
Comment (sunverwerth) [inline Userland/Libraries/LibGL/Lighting.cpp:12]: I think all those extern declarations can be removed.

**Review by sunverwerth — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibGL/Lighting.cpp (hunk 1 above)
Comment (GMTA) [reply Userland/Libraries/LibGL/Lighting.cpp]: They can! And they are :-)

**Comment by sunverwerth** [comment]
Generally I like these changes. Does the split follow any easy to follow guildelines? Are those files separated according to their chapter in the GL spec?

**Comment by GMTA** [comment]
> Does the split follow any easy to follow guildelines? Are those files separated according to their chapter in the GL spec?

No, I based the split purely on what seemed sensible to me based on a base functionality (e.g. stencil, texturing) and what functions are directly associated with that.

Do you think it would be possible at all to use the OpenGL specs for some ordering? I got the feeling that the different versions of the spec wildly differ in their arrangement of functionalities.

**Comment by sunverwerth** [comment]
> I got the feeling that the different versions of the spec wildly differ in their arrangement of functionalities.
Yeah that might be true.

**Pull request merged**